server/test/repository/ProgramRepositoryTest.java

```diff
@@ -846,6 +849,116 @@ public void getApplicationsForAllProgramVersions_sortedBySubmitDate() {
     assertThat(paginationResult.getPageContents().get(0).getApplicant()).isEqualTo(applicantThree);
   }
 
+  @Test
+  public void getApplicationsForAllProgramVersions_sortedAndPagedBySubmitTime() {
+    ApplicantModel applicantOne =
+        resourceCreator.insertApplicantWithAccount(Optional.of("one@example.com"));
+    ProgramModel originalVersion = resourceCreator.insertActiveProgram("test program");
+
+    ApplicationModel applicationOne =
+        resourceCreator.insertActiveApplication(applicantOne, originalVersion);
+
+    ProgramModel nextVersion = resourceCreator.insertDraftProgram("test program");
+    resourceCreator.publishNewSynchronizedVersion();
+
+    ApplicantModel applicantTwo =
+        resourceCreator.insertApplicantWithAccount(Optional.of("two@example.com"));
+    ApplicantModel applicantThree =
+        resourceCreator.insertApplicantWithAccount(Optional.of("three@example.com"));
+    ApplicationModel applicationTwo =
+        resourceCreator.insertActiveApplication(applicantTwo, nextVersion);
+    ApplicationModel applicationThree =
+        resourceCreator.insertActiveApplication(applicantThree, nextVersion);
+
+    /* Set the submit date such that the results come back in this order: 2, 1, 3 */
+    applicationThree.setSubmitTimeToNow();
+    applicationThree.save();
+    applicationOne.setSubmitTimeToNow();
+    applicationOne.save();
+    applicationTwo.setSubmitTimeToNow();
+    applicationTwo.save();
```

Comment:
Do these come through as different times?

If so, it might be good to have a test that also tests when a set of multiple applications come through at the same time, with the page boundary in the middle of that set, to make sure the ID ordering is working correctly too.

Reply:
yes, these sets different times. 

I didn't add a test for the same specific time, because that test exists all over the code right now .. just about every test that exists right now neglects to set a submit time , and so you end up with the applications sorted by ID ..